### ISSUE TITLE
GetMetadataAsync() method stopped returning after upgrade to version 5.6.2

### ISSUE THREAD

**Issue by kgenov** [simple-odata-client/Simple.OData.Client#587]
Hi Vagif,

I was using version 4.29.0  and decided to upgrade to latest stable 5.6.2. But at that point the following code stopped working:

            ODataClientSettings settings = new ODataClientSettings(new Uri(this.PortalService.ServiceBaseUrl), credentials);
            settings.IncludeAnnotationsInResults = true;
            settings.PayloadFormat = ODataPayloadFormat.Atom;
            settings.IgnoreUnmappedProperties = true;
            settings.BeforeRequest = BeforeRequest;
            settings.AfterResponse = AfterResponse;
            this.client = new ODataClient(settings);
            var metadata = this.client.GetMetadataAsync().Result; // stops here
            this.metadataParser = new MetadataCsdlSemanticsModelParser(metadata);

Basically, the code never makes it to the last line, it gets to the line

            var metadata = this.client.GetMetadataAsync().Result;

and never goes to the next line.

In Fiddler I can see that the metadata is requested and returned properly from the server with status 200 OK. Also, the AfterResponse handler is fired and in it I also get status 200 OK.

All projects in my solution target .Net 4.6.2

If I go back to version 4.29.0 it all works.

Can you please give some advise what else to look at?

Thanks!
Kiril

**Comment by object** [comment]
Hi Kirill,
Versions 4 and 5 are very different. V5 is based on .NET Standard. I am sure your problem is due to not properly awaiting async call. Try using await instead of calling .Result or .Wait.

**Comment by kgenov** [comment]
Hi Vagif,

Thank you very much for your prompt response and for pointing me in the right direction!

It appears to be a deadlock problem. My colleague proposed this solution and it worked perfectly:

`var metadata = Task.Run(() => this.client.GetMetadataAsync()).Result;`

No more deadlocks!

Thanks!
Kiril

**Comment by object** [comment]
Good to know!
